pull_request_template.md

```diff
@@ -9,45 +9,62 @@ Closes: #XXX
 
 ## What is the purpose of the change
```

Comment:
```suggestion
## Context and purpose of the change
```

It seems like lots of PRs have recently been adding additional sections explaining things, let's change this title and consolidate those sections here

Reply:
done